flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/LoopFlowUtil.java

```diff
@@ -0,0 +1,154 @@
+
+import com.farao_community.farao.balances_adjustment.util.CountryArea;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_file.CracFile;
+import com.farao_community.farao.data.flowbased_domain.DataMonitoredBranch;
+import com.farao_community.farao.data.flowbased_domain.DataPtdfPerCountry;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationParameters;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationProvider;
+import com.farao_community.farao.flowbased_computation.FlowBasedComputationResult;
+import com.farao_community.farao.flowbased_computation.glsk_provider.GlskProvider;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Network;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Loop flow util is used
+ * - for loop flow calculation (used in search tree rao for CCR Core), and
+ * - for Min RAM adjustment calculation
+ * @author Pengbo Wang {@literal <pengbo.wang at rte-international.com>}
+ */
+public final class LoopFlowUtil {
+    private LoopFlowUtil() {
+    }
+
+    /**
+     * calculate loop flow
+     * input CracFile
+     * This function could be removed when Flowbased computation no longer use CracFile but Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
+                                                         CracFile cracFile,
+                                                         GlskProvider glskProvider,
+                                                         List<String> countries,
+                                                         ComputationManager computationManager,
+                                                         FlowBasedComputationParameters flowBasedComputationParameters
+    ) {
+        //Call flowbased computation on given glskProvider
+        FlowBasedComputationProvider flowBasedComputationProvider = new FlowBasedComputationImpl();
+        FlowBasedComputationResult flowBasedComputationResult = flowBasedComputationProvider.run(network,
+                cracFile, glskProvider, computationManager, network.getVariantManager().getWorkingVariantId(), flowBasedComputationParameters)
+                .join();
+
+        Map<String, Double> frefResults = frefResultById(flowBasedComputationResult); //get reference flow
+        Map<String, Map<String, Double>> ptdfResults = ptdfResultById(flowBasedComputationResult); // get ptdf
+        Map<String, Double> referenceNetPositionByCountry = getRefNetPositionByCountry(network, countries); // get Net positions
+        return buildLoopFlowsFromResult(flowBasedComputationResult, frefResults, ptdfResults, referenceNetPositionByCountry);
+    }
+
+    /**
+     * calculate loop flow
+     * input Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
```

Comment:
In this method can you remove glskProvider and countries that you seek directly in the crac extension. And then you can create the LoopFlowExtensions directly and add it to the cnec. So you modify the crac directly. 
And so maybe rename it addLoopFlowDataToCrac or something like that.

Reply:
What you described is done in this function: calculateLoopFlowConstraintAndUpdateAllCnec() of SearchTreeRao.

The calculateLoopFlows keeps its signatures because in MinimumRAM we used different GlskProvider. It is in the beginning of calculateAMR() of MinRamAdjustment